src/mui/detail/Edit.js

```diff
@@ -28,12 +26,8 @@ export class Edit extends Component {
     componentWillReceiveProps(nextProps) {
         if (this.props.data !== nextProps.data) {
             this.setState({ record: nextProps.data }); // FIXME: erases user entry when fetch response arrives late
-            if (this.fullRefresh) {
-                this.fullRefresh = false;
-                this.setState({ key: this.state.key + 1 });
-            }
         }
-        if (this.props.id !== nextProps.id) {
+        if (this.props.id !== nextProps.id || nextProps.version !== this.props.version) {
```

Comment:
The refresh button changes a version. If we see that the version was changed we fetch a new data about the record. Did I miss something?